src/Symfony/Component/Security/Core/Authorization/Voter/SwitchRoleVoter.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization\Voter;
+
+use Symfony\Component\Security\Core\Authentication\Token\SwitchUserToken;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AccessDecisionManager;
+
+/**
+ * SwitchRoleVoter votes if we check the impersonator roles.
+ *
+ * @author Fabien Papet <fabien.papet@gmail.com>
+ */
+class SwitchRoleVoter implements VoterInterface
+{
+    /**
+     * @var AccessDecisionManager
```

Comment:
Can be removed since this is inferred from the controller.

Reply:
from the constructor* right ?

Reply:
Yes this auto correction is amazing some time :D